### PR TITLE
Implement react-native-reorderable-list

### PR DESCRIPTION
**Pull request by Jon-edge**
Notes:
- Did not include the patch package from the original rn-sortable-listview integration
- TODO: Unable to get the useSceneScrollHandler to work with the SortableListView

May need to go with the new library route.

### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [x] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

### CODE REVIEW

**Comment by swansontec** [comment]
Was this tested on iOS? We should definitely do that before merging.

File: jestSetup.js [hunk 1]
```diff
@@ -254,3 +254,8 @@ jest.mock('react-native-device-info', () => {
     getVersion: jest.fn()
   }
 })
+
+jest.mock('react-native-reorderable-list', () => ({
+  ...jest.requireActual('react-native-reorderable-list'),
+  useReorderableDrag: () => jest.fn(),
+}))
```
Comment (swansontec) [inline jestSetup.js:261]: Missing newline in file.

File: src/state/SceneScrollState.tsx [hunk 2]
```diff
@@ -113,3 +113,37 @@ export const useSceneScrollHandler = (): SceneScrollHandler => {
 
   return handler
 }
+
+export const useSceneScrollWorkletHandler = () => {
```
Comment (swansontec) [inline src/state/SceneScrollState.tsx:117]: Please add a comment to this to explain why this is different than the one above.

File: src/components/themed/WalletListSortable.tsx [hunk 3]
```diff
@@ -23,39 +20,46 @@ interface Props {
...
-    for (let i = 0; i < walletOrder.length; ++i) {
-      const walletId = walletOrder[i]
+    for (let i = 0; i < newOrder.length; ++i) {
+      const walletId = newOrder[i]
       keyStates[walletId] = { sortIndex: i }
     }
     account.changeWalletStates(keyStates).catch(error => showError(error))
   })
 
+  const keyExtractor = (item: string) => item
```
Comment (swansontec) [inline src/components/themed/WalletListSortable.tsx:45]: I think this needs a useHandler or useCallback as well.

**Review by swansontec — CHANGES_REQUESTED** [review]
Small changes needed, but this looks good.

File: src/components/themed/WalletListSortable.tsx (hunk 3 above)
Comment (Jon-edge) [reply src/components/themed/WalletListSortable.tsx:45]: Using a useHandler somehow broke the feature

Comment (Jon-edge) [reply src/components/themed/WalletListSortable.tsx:45]: Figured it out, needed to use `useCallback`

**Review by swansontec — APPROVED** [review]

**Pull request merged**